### ISSUE TITLE
getTransactions RPC call not returning any transactions

### ISSUE THREAD

**Issue by jared201** [cryptonotefoundation/cryptonote#170]
quick question, I was able to mine, transfer, and get balance, but when I call JSON RPC for an specific address, it's returning an empty transactions arrays

{"id":"test","jsonrpc":"2.0","result":{"items":[{"blockHash":"fd2a1a0afa4f75da80c4ab0cc22a692c41197a2fa873daee0cf69897c5182de8","transactions":[]},{"blockHash":"fcd854de0d0ad06edc75b722d04767286d3c728188645dededf1fc171f96af81","transactions":[]},{"blockHash":"2cf4f30365b4b5b095fcddcab7d20043ed9abdbf73ce347deced192aa8e41dbc","transactions":[]},{"blockHash":"a458156e5b573ed0b63100056f14a0ac59ed8c4d23538011af18117702892d02","transactions":[]},{"blockHash":"ae9a1ebd2c2a3d3d7fdfd29429dd1ea5b74facb00fc726ab228506042d61e0af","transactions":[]},{"blockHash":"22ab459eae2194aeb327abb000f69fa2ed900c21e383b7e4d8a4a64d2864076f","transactions":[]},{"blockHash":"015635f9c3b101cdc62ca53fa4f4e09308d7b6cc47ecf24f0b4bc12add9c2603","transactions":[]},{"blockHash":"2bd43c1f7ecdc2669fae42373b4a1af2cb39684fdc18293fdc9c117934e8cad8","transactions":[]},{"blockHash":"cfeed1fe92bbf2318dde3fc0d13af379a4f20253cb92a2d899fadde7b4b785fe","transactions":[]},{"blockHash":"172d67ca309ccb71e94ce773b39c1dab757f8ed7a05e7f77896882449ae7ef74","transactions":[]},{"blockHash":"fd56bbb8f9166c6ac1f52c60a3bd0a330dbb527c546a18778d52714700a750fa","transactions":[]},{"blockHash":"a97f2df687b8df938f097791a47737a20a616003e0c9a736df7d3aeb79dec352","transactions":[]},{"blockHash":"5779c8c0fc26d2999f5a520057aaf84e9cd97845e8f3fb0ff6a5400aff308bbd","transactions":[]},{"blockHash":"3c9fce3a94347f95192325fc302f9ee3bd74e4bd2ffff4c597289acc1deb0ebd","transactions":[]},{"blockHash":"8e5a4658aee238846d62f18b9e8881114c1c7ddaea83521005fa0bc7de1221ab","transactions":[]},{"blockHash":"be26611392a36aaef4be2d6ad1cd6bce489f7405c2e59a56a8f988d6aeaa9a96","transactions":[]},{"blockHash":"5da55fc129b83f5abddb3071d3d060b48588778e0830b2e5ad67dddca1191709","transactions":[]},{"blockHash":"c094d1b7bd42a73fc7f7c85c677820db7644ea5d799c4ddf4d2a54f250acfa9e","transactions":[]},{"blockHash":"099876ba5a6b834c4a83bf8a122bfe66cc55b09eae1f9381f0ceff92620deecf","transactions":[]},{"blockHash":"b31ac317aa8dce6dac2d374d4fc131682397acc269cf9406acf11043971014b8","transactions":[]}]}}

Next i'm going to try is to set anonymity to 1 if it changes anything

**Comment by ghost** [comment]
Hi Jared,
I know it's not concerning this topic but i saw your post on issue #152 and i would like to know how you solved the multithreading issue with the miner? I didn't see any clear explanation in the post.
You can contact me also on djouanen[@]gmail.com

**Comment by SoraKohaku** [comment]
i think he already fixed~

**Comment by jared201** [comment]
@Deedjee I was running with Ubuntu 14.04 for the main node, I build a second node on 16.04 using all the latest fixes from different repos, I left the main node as the coin daemon and run all the mining from the second node, there's no point fixing the main node's issue now, to make it quick you can fork my repo

**Comment by SoraKohaku** [comment]
ahaha nice done. but sometimes here files not upgraded and some unnecesary
used.~

**Comment by bomb-on** [comment]
@jared201 Did you solve the issue of not getting any transactions in response maybe?

**Comment by SoraKohaku** [comment]
trans will unlock after some block will confirm

On Feb 13, 2018 10:04 PM, "bomb-on" <notifications@github.com> wrote:

@jared201 <https://github.com/jared201> Did you solve the issue of not
getting any transactions in response maybe?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/cryptonotefoundation/cryptonote/issues/170#issuecomment-365260806>,
or mute the thread

**Comment by bomb-on** [comment]
@FndNur1Labs Yeah, I know that, but I'm troubled with something different here... Trying to use walletd JSON RPC API... This is the sample command:

`curl -X POST http://127.0.0.1:8070/json_rpc -H 'Cache-Control: no-cache' -H 'Content-Type: application/json' -d '{"method":"getTransactions","params":{"firstBlockIndex":90007,"blockCount":1}}'
`

Testing this on 2 machines. Both running coin daemon and walletd with same flags and ports and everything else. Both have fully synced blockchain which is now near block 100k...

Response from the daemon on one machine:

```
{
  "jsonrpc": "2.0",
  "result": {
    "items": [
      {
        "blockHash": "732b37e627eb81f2a49ed340b2e2619bf5fd1fa37ad60d82845bb7388b1cfe23",
        "transactions": []
      }
    ]
  }
}```

Response from the daemon on the other machine:

```{
  "jsonrpc": "2.0",
  "result": {
    "items": [
      {
        "blockHash": "732b37e627eb81f2a49ed340b2e2619bf5fd1fa37ad60d82845bb7388b1cfe23",
        "transactions": [
          {
            "amount": 9057237,
            "blockIndex": 90007,
            "extra": "013d8a389f5430ebc07622fa2bc456caa05b7e9be8610b4b41ee0569821b4e21aa0208000000000a4f2dbc",
            "fee": 0,
            "isBase": true,
            "paymentId": "",
            "state": 0,
            "timestamp": 1518193326,
            "transactionHash": "30de50736c4f8c5bf77430090fa7928cc886865cc83a3ce59bc723026a0c2a2d",
            "transfers": [
              {
                "address": "XP1JoKorjgT7ifXK3M5yLEQH2nRtNWvUUwc1YHpeeD4ZESb1kD7wLLAKYExoEcAbDjnA6vxZQ5zzKohfNhnxHmr2qaL6HTUEq",
                "amount": 9057237,
                "type": 0
              }
            ],
            "unlockTime": 90021
          }
        ]
      }
    ]
  }
}```

This block is more than confirmed and matured, I'm pretty sure about it, but I just cannot explain to myself why is this happening. To make it more fun, the daemon on the first machine always responds with empty array of transaction, for literally every single block in blockchain :/

On both machines, if I check the status with getStatus() I can see something like this:

```{
  "jsonrpc": "2.0",
  "result": {
    "blockCount": 94061,
    "knownBlockCount": 94061,
    "lastBlockHash": "5ec49e2a4201f1c3cad8fc79a2abec8a794d8fd24136962cb83af8781ed584dc",
    "peerCount": 23
  }
}```

What is happening with my daemons?? :(

**Comment by sreerajs** [comment]
@jared201  Hey has your repo fixed this? or are you still running deamon and mining on diffrent nodes?
You mentioned your repo had fixes from others. I was trying to build something on cryptonote and the official repo seems to be a lot buggy. I was wondering what all changes you had to make to source code for making it development ready. New to this, a little help appreciated. Thanks!

**Comment by SoraKohaku** [comment]
go to my repo we will ready for Adventure people...;)
